Funcky.SourceGenerator/OrNoneFromTryPatternGenerator.cs

```diff
@@ -40,69 +37,134 @@ private static IncrementalValueProvider<ImmutableArray<MethodPartial>> GetOrNone
+            .WithType(GenerateTypeSyntax(parameter.Type))
+            .WithDefault(GetParameterDefaultValue(parameter));
+
+    private static TypeSyntax GenerateTypeSyntax(ITypeSymbol type)
+    {
+        var parsedType = ParseTypeName(type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.AddMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier)));
+        return type.NullableAnnotation is not NullableAnnotation.None
+            ? parsedType
+            : parsedType
+                .WithLeadingTrivia(Trivia(NullableDirectiveTrivia(Token(SyntaxKind.DisableKeyword), true)))
+                .WithTrailingTrivia(Trivia(NullableDirectiveTrivia(Token(SyntaxKind.EnableKeyword), true)));
     }
 
-    private static string GetMethodValue(Compilation compilation, MethodDeclarationSyntax method, AttributeSyntax attribute)
-        => attribute.ArgumentList?.Arguments.Count > 1 && attribute.ArgumentList?.Arguments[1].Expression is { } descriptionExpr
-            ? compilation
-                .GetSemanticModel(method.SyntaxTree)
-                .GetConstantValue(descriptionExpr)
-                .ToString()
-            : throw new Exception("Method value on attribute missing.");
-
-    private static TypeSyntax GetTypeValue(Compilation compilation, MethodDeclarationSyntax component, AttributeSyntax attribute)
-        => attribute.ArgumentList?.Arguments.Count > 0 && attribute.ArgumentList?.Arguments[0].Expression is TypeOfExpressionSyntax iconExpr
-            ? iconExpr.Type
-            : throw new Exception("Type value on attribute missing.");
-
-    private static string? GetClassName(SyntaxNode methodDeclaration)
-        => methodDeclaration
-            .TryGetParentSyntax<ClassDeclarationSyntax>()
-            ?.Identifier
-            .ToString();
-
-    private static string? GetNamespaceName(SyntaxNode methodDeclaration, Compilation compilation)
+    private static string GetNamespaceName(SyntaxNode methodDeclaration, Compilation compilation)
         => compilation.GetSemanticModel(methodDeclaration.SyntaxTree)
             .GetDeclaredSymbol(methodDeclaration)?
             .ContainingNamespace
-            .ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted));
+            .ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted))
+                ?? throw new InvalidOperationException("Unable to detect containing namespace");
 
-    private static MethodDeclarationSyntax CreateMethodImplementation(MethodDeclarationSyntax methodDeclaration, string methodName, TypeSyntax typeSyntax)
-        => (MethodDeclarationSyntax)new OrNoneFromTryPatternRewriter(methodName, typeSyntax)
-            .Visit(methodDeclaration);
+    private static string GetMethodName(ITypeSymbol type, IMethodSymbol method)
```

Comment:
You could widen the type to ISymbol.

Reply:
I think it's better as is, since it protects against passing the arguments in the wrong order.